scripts: bench: switch back to the new branch

RaduBerinde opened the pull request:
The bench script leaves the old branch checked out. Normally we are
working on the new branch, so switch back to that.

Release note: None

RaduBerinde committed:
scripts: bench: switch back to the new branch

The bench script leaves the old branch checked out. Normally we are
working on the new branch, so switch back to that.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/23314)
<!-- Reviewable:end -->

jordanlewis commented:
LGTM, this bothers me too.

RaduBerinde committed:
scripts: bench: use current branch for new

Improving bench to use the current branch if a "new" branch isn't
specified.

Release note: None

RaduBerinde commented:
I added another commit.

#### scripts: bench: use current branch for new

Improving bench to use the current branch if a "new" branch isn't
specified.

RaduBerinde commented:
PTAL at the second commit. Thanks!
